### PR TITLE
rsx/common/d3d12: Support for texture 1d too.

### PR DESCRIPTION
**Pull request by vlj**
They are used in after burner climax

### CODE REVIEW

**Comment by vlj** [comment]
Rebased and fill normalised coord field in get_current_fragment_program.

File: rpcs3/Emu/RSX/D3D12/D3D12FragmentProgramDecompiler.cpp [hunk 1]
```diff
@@ -118,7 +118,16 @@ void D3D12FragmentDecompiler::insertConstants(std::stringstream & OS)
 
 	for (const ParamType &PT : m_parr.params[PF_PARAM_UNIFORM])
 	{
-		if (PT.type == "sampler2D")
+		if (PT.type == "sampler1D")
+		{
+			for (ParamItem PI : PT.items)
```
Comment (danilaml) [inline rpcs3/Emu/RSX/D3D12/D3D12FragmentProgramDecompiler.cpp]: `const &`?

File: rpcs3/Emu/RSX/D3D12/D3D12FragmentProgramDecompiler.cpp [hunk 2]
```diff
@@ -118,7 +118,16 @@ void D3D12FragmentDecompiler::insertConstants(std::stringstream & OS)
...
+			{
+				size_t textureIndex = atoi(PI.name.data() + 3);
+				OS << "Texture1D " << PI.name << " : register(t" << textureIndex + 16 << ");" << std::endl;
+				OS << "sampler " << PI.name << "sampler : register(s" << textureIndex << ");" << std::endl;
+			}
+		}
+		else if (PT.type == "sampler2D")
 		{
 			for (ParamItem PI : PT.items)
```
Comment (danilaml) [inline rpcs3/Emu/RSX/D3D12/D3D12FragmentProgramDecompiler.cpp]: And here.

**Comment by vlj** [comment]
rebased

**Pull request merged**